### PR TITLE
Allow running the app as a single artifact

### PR DESCRIPTION
**Pull request by textbook**
* A short explanation of the proposed change:

    Allow the app to run as a single artifact, with the backend Rails app serving the built React frontend. This topology is used for the tests, the packaged deployment is still to two separate apps (the only change here is that the API moves from `server-domain.tld/` to `server-domain.tld/api`, which **may break some workflows** if people are accessing the API directly).

* An explanation of the use cases your change solves

    Being able to deploy a single artifact allows us to make progress to #105 and opens up simplified deployment options across all supported platforms.

* Links to any other associated PRs

    N/A

* [x] I have reviewed the [contributing guide](https://github.com/pivotal/postfacto/blob/master/CONTRIBUTING.md)

* [x] I have made this pull request to the `master` branch

* [x] I have run all the tests using `./test.sh`

* [x] I have added the [copyright headers](https://github.com/pivotal/postfacto/blob/master/license-header.txt) to each new file added

* [x] I have given myself credit in the [humans.txt](https://github.com/pivotal/postfacto/blob/master/humans.txt) file (assuming I want to)

### CODE REVIEW

File: deployment/mixpanel.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Comment (crswty) [inline deployment/mixpanel.sh]: Should this be a separate PR?

File: web/src/components/retro-show/show_retro_page.jsx [hunk 2]
```diff
@@ -319,7 +331,7 @@ class ShowRetroPage extends React.Component {
 
     return (
       <span>
-        <RetroWebsocket url={config.websocket_url} retro_id={retroId} websocketRetroDataReceived={this.props.websocketRetroDataReceived}/>
+        <RetroWebsocket url={websocketUrl(config)} retro_id={retroId} websocketRetroDataReceived={this.props.websocketRetroDataReceived}/>
```
Comment (crswty) [inline web/src/components/retro-show/show_retro_page.jsx]: Can we not just use a relative URL for the websocket? Is this a PWS thing?

Comment (seadowg) [reply web/src/components/retro-show/show_retro_page.jsx]: Depending on the setup it will often run on a different port

**Comment by crswty** [comment]
I'm taking a glance at this but I'm really out of date with how the packaging works, also not not confident understanding the API changes. Is someone else in a better position to actually review? @seadowg  or @tom025 maybe?

**Comment by crswty** [comment]
@seadowg Are you able to look at this?

File: deployment/mixpanel.sh (hunk 1 above)
Comment (seadowg) [reply deployment/mixpanel.sh]: Yeah I think so.

**Comment by textbook** [comment]
@crswty @seadowg I've pulled the Mixpanel bit out into #190 

Yes, it works by copying the FE assets into `./api/client`, then having Rails serve static assets from both that and `./api/public/`. It moves the actual API endpoints down to `/api` to avoid conflicts.

**Comment by crswty** [comment]
Only outstanding thing I can think of is upgrade path, is it upgradable in place or should we cut a major version?

**Comment by textbook** [comment]
@crswty challenge right now is the scripted Heroku upgrade, actually caused by dependency upgrades: https://github.com/pivotal/postfacto/issues/183. We can upgrade the old web instance to the new API-and-web instance to avoid this, but then have to do awkward things to get the data over and be able to delete the old API instance (see https://stackoverflow.com/q/57498273/3001761).

**Comment by crswty** [comment]
So if I understand, there is still the option to deploy two apps of this blocks you and an ability to manually migrate if you choose to?

If that's correct my sense is that's fine if we document it, what do you think?

File: deployment/mixpanel.sh (hunk 1 above)
Comment (textbook) [reply deployment/mixpanel.sh]: Got merged as #190
